### ISSUE TITLE
Patch: correct show information in FreeBSD - CPU, RAM, uptime, network (+MTU)

### ISSUE THREAD

**Issue by VVD** [nextcloud/serverinfo#203]
### Steps to reproduce
1. Open http://my-nexcloud.my.domain/index.php/settings/admin/serverinfo
2. Look at CPU, RAM, uptime, network information

### Expected behaviour
See correct information

### Actual behaviour
No this information

### Server configuration

**Operating system**:
FreeBSD 12.1-RELEASE-p3 amd64

**Web server:**
Apache 2.4.43

**Database:**
PostgreSQL 12.2

**PHP version:**
7.4.4

**Nextcloud version:** (see Nextcloud admin page)
18.0.3.0

Don't think other information are necessary.
Patch is here: [serverinfo.diff.txt](https://github.com/nextcloud/serverinfo/files/4496997/serverinfo.diff.txt)

This patch fixes #172.

P.S. I'm not a php programmer, but this part of core is obvious. :-)

**Comment by VVD** [comment]
OS name doesn't show too.
nextcloud/apps/serverinfo/templates/settings-admin.php:
```
<?php p($l->t('Operating System'));?>:
<span class="info"><?php p($_['osname']);?></span>
```
Can't find there 'osname' filled…

I can create method in DefaultOs.php.
For FreeBSD it's "uname -mrs", for Linux it can be same "uname -mrs".

**Comment by VVD** [comment]
Disk size show with default block size 512 bytes instead of 1KB.
This patch work fine for both FreeBSD and Linux:
nextcloud/apps/serverinfo/lib/OperatingSystems/DefaultOs.php:
```
-                $data  = shell_exec('df -TP');
+                $data  = shell_exec('df -TPk');
```

Swap size show with default block size 512 bytes too:
```
$ BLOCKSIZE= swapinfo 
Device          512-blocks     Used    Avail Capacity
/dev/da1           8388608   214560  8174048     3%
$ BLOCKSIZE=K swapinfo 
Device          1K-blocks     Used    Avail Capacity
/dev/da1          4194304   107280  4087024     3%
$ swapinfo -k 
Device          1K-blocks     Used    Avail Capacity
/dev/da1          4194304   107280  4087024     3%
$ swapinfo -m
Device          1M-blocks     Used    Avail Capacity
/dev/da1             4096      103     3992     3%
$ swapinfo -g
Device          1G-blocks     Used    Avail Capacity
/dev/da1                4        0        3     3%
```
Nextcloud show this information:
SWAP: Всего: 7,8 GB - Текущая нагрузка: 209,5 MB
Patch for nextcloud/apps/serverinfo/lib/SystemStatistics.php is:
```
-      exec("/usr/sbin/swapinfo", $return, $status);
+      exec("/usr/sbin/swapinfo -k", $return, $status);
```


There is sysctl variable vm.swap_total in bytes:
```
$ sysctl -n vm.swap_total
4294967296
$ sysctl -dn vm.swap_total
Total amount of available swap storage.
```

I can add all these patches to 1st one.

**Comment by VVD** [comment]
> OS name doesn't show too.
> nextcloud/apps/serverinfo/templates/settings-admin.php:
> 
> ```
> <?php p($l->t('Operating System'));?>:
> <span class="info"><?php p($_['osname']);?></span>
> ```
> 
> Can't find there 'osname' filled…
> 
> I can create method in DefaultOs.php.
> For FreeBSD it's "uname -mrs", for Linux it can be same "uname -mrs".

It was fixed already in master branch:
```
PHP_OS . ' ' . php_uname('r') . ' ' . php_uname('m');
```

**Comment by kesselb** [comment]
Looks good to me. Please copy `DefaultOs.php` to `FreeBSD.php` and add all freebsd specific code there. Then add the switch (DefaultOs or FreeBSD) to `Os.php` and submit a pull request.

**Comment by VVD** [comment]
> Looks good to me. Please copy `DefaultOs.php` to `FreeBSD.php` and add all freebsd specific code there. Then add the switch (DefaultOs or FreeBSD) to `Os.php`
```
use OCA\ServerInfo\OperatingSystems\DefaultOs;
use OCA\ServerInfo\OperatingSystems\FreeBSD;
…
if (PHP_OS === 'FreeBSD') {
        $this->backend = new FreeBSD();
} else {
        $this->backend = new DefaultOs();
}
```
This way?
> and submit a pull request.

I don't know how to do a pull request… But I can attach all files here.

**Comment by leManu** [comment]
> I don't know how to do a pull request… But I can attach all files here.

1. Fork this repo to a personal one (click the **Fork** button on the upper right of the page https://github.com/nextcloud/serverinfo)
2. Clone your personal repo to your machine, and make your modifications
3. Commit and Push the modifications to your personal repo
4. Follow this documentation to create a pull request : https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork

**Comment by VVD** [comment]
Patch (diff -urN) is here:
[serverinfo.diff.txt](https://github.com/nextcloud/serverinfo/files/4504535/serverinfo.diff.txt)
All changed files are here:
[serverinfo.tar.gz](https://github.com/nextcloud/serverinfo/files/4504541/serverinfo.tar.gz)

>1. Fork this repo to a personal one
Plz, no…
Can you just get patch?

**Comment by VVD** [comment]
Is it possible to see this patch (or parts thereof) in the main tree?
We can split it into several independent patches.

**Comment by VVD** [comment]
Any possibility to see this patch in NextCloud 19?

**Comment by tflidd** [comment]
> > 1. Fork this repo to a personal one
> 
> Plz, no…
> Can you just get patch?

It's easier if you do it. In case there are change requests, you can directly do this. Cloning it into your personal github account and push the modifications into a branch on your personal repo isn't that difficult since you managed to do all the coding.



> Any possibility to see this patch in NextCloud 19?

Sure, once it is tested and then merged here, it gets out into the app (even in NC 18 I suppose).

**Comment by VVD** [comment]
If I fork all repos I wrote a single small patch - I'll have too much repos.

**Comment by kesselb** [comment]
Fixed by #210 for Nextcloud 20.
